### DISCUSSION TITLE
lua config zone

### DISCUSSION THREAD

**Discussion by mayeter** [PowerDNS/pdns#14427] [Q&A] [answered]
- Program: Authoritative 
 - Issue type: Feature request

### Short description

Hello, I would like to include lua config records that are created under different zone. Like it is done [here](https://doc.powerdns.com/authoritative/lua-records/index.html#advanced-topics) but if include works between zones, that would be great to simplify configs.

### Usecase

I am working in a multi datacenter environment. I will use view() function to return different ip addresses based on client. Since I have so many local zones, I would like to create config once and include it from other zones.

### Description

I've been able to create config records as described in your documentation (the one I linked above). It works like a charm 👏 
I have powerdns authoritative name servers (v 4.8.0, sqlite3 backend) in my multi data center infra. view() function works as I expected and I can include subnet informations as variables from this config record. I can work with that but if I could create a seperate zone just to hold these variables and include them between zones (since I manage ~40 zones) it would be great for my operational workload. Is there any option I could use at the moment that I missed in your documentations 🙏

**Comment by Habbie** [comment]
> Is there any option I could use at the moment that I missed in your documentations

you can use a file (although that is not convenient for everybody), or [`dblookup`](https://doc.powerdns.com/authoritative/lua-records/functions.html?highlight=dblookup#dblookup) to build a fancier `include` yourself

**Reply by mayeter** [reply]
hello, could you direct me about loading file part of the documentation, I would like to give it a shot. dblookup is not looking convenient for my purpose or I am mis-interprating the docs.

**Reply by Habbie** [reply]
[`loadfile`](https://lua.org/manual/5.1/manual.html#pdf-loadfile)

[`require`](https://lua.org/manual/5.1/manual.html#pdf-require)

**Reply by mayeter** [reply]
hello habbie, best I could do is:

`A ";local o = dofile('/var/lib/powerdns/commom-lua.d/01-ipmap.lua') return view({{o.site1, {'10.1.1.1'} } , { o.site2, {'10.2.1.1'} } , { {'0.0.0.0/0'}, {'10.0.0.1'} }} ) "`

with the 01-ipmap.lua file:

```
local subnets = {
        site1 = {'10.1.0.0/16', '10.11.0.0/16'},
        site2 = {'10.2.0.0/16'},
        site3 = {'10.3.0.0/16'}
}

return subnets
```

this works but I would like to achive something even simpler(i am trying to make it self service for some privileged teams), it would be great if I could register site1,site2 etc as global variables and just use them in lua records without importing them, is it possible? (today or in the future)

**Reply by Habbie** [reply, marked as answer]
this was requested before in https://github.com/PowerDNS/pdns/issues/11676 and there's code in https://github.com/PowerDNS/pdns/pull/11678, which no doubt some day will be finished and merged

**Reply by mayeter** [reply]
yeah I saw this one but it seems inactive since almost a year, but if you say so I am waiting for it then. thank you for your time 🙏

**Comment by airsmon** [comment]
> * Program: Authoritative
> * Issue type: Feature request
> 
> ### Short description
> Hello, I would like to include lua config records that are created under different zone. Like it is done [here](https://doc.powerdns.com/authoritative/lua-records/index.html#advanced-topics) but if include works between zones, that would be great to simplify configs.
> 
> ### Usecase
> I am working in a multi datacenter environment. I will use view() function to return different ip addresses based on client. Since I have so many local zones, I would like to create config once and include it from other zones.
> 
> ### Description
> I've been able to create config records as described in your documentation (the one I linked above). It works like a charm 👏 I have powerdns authoritative name servers (v 4.8.0, sqlite3 backend) in my multi data center infra. view() function works as I expected and I can include subnet informations as variables from this config record. I can work with that but if I could create a seperate zone just to hold these variables and include them between zones (since I manage ~40 zones) it would be great for my operational workload. Is there any option I could use at the moment that I missed in your documentations 🙏

Hello, may I ask how you configure it here? I also have the same requirement, to return different resolution information based on the requested IP in the inaccessible area
